src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
+            if raise_on_exception:
+                raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+            # more return codes should be added as appropriate
+        else:  # verbose diagnostic log
+            self.composite_logger.log_debug('[TDNF] Invoked package manager. [Command={0}][Code={1}][Output={2}]'.format(command, str(code), str(out)))
+        return out, code
+
+    # region Classification-based (incl. All) update check
+    def get_all_updates(self, cached=False):
+        """Get all missing updates"""
+        self.composite_logger.log_verbose("[TDNF] Discovering all packages...")
+        if cached and not len(self.all_updates_cached) == 0:
+            self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(cached), len(self.all_updates_cached)))
+            return self.all_updates_cached, self.all_update_versions_cached  # allows for high performance reuse in areas of the code explicitly aware of the cache
+
+        out = self.invoke_package_manager(self.tdnf_check)
+        self.all_updates_cached, self.all_update_versions_cached = self.extract_packages_and_versions(out)
+
+        self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(False), len(self.all_updates_cached)))
+        return self.all_updates_cached, self.all_update_versions_cached
+
+    def get_security_updates(self):
+        """Get missing security updates"""
+        self.composite_logger.log_verbose("[TDNF] Discovering 'security' packages...")
+        self.composite_logger.log_warning("Classification based categorization of patches is not available in Azure Linux as of now.")
+        security_packages, security_package_versions = [], []
+        self.composite_logger.log_debug("[TDNF] Discovered 'security' packages. [Count={0}]".format(len(security_packages)))
+        return security_packages, security_package_versions
+
+    def get_other_updates(self):
+        """Get missing other updates"""
+        self.composite_logger.log_verbose("[TDNF] Discovering 'other' packages...")
+        other_packages = []
+        other_package_versions = []
+
+        all_packages, all_package_versions = self.get_all_updates(True)
+        security_packages, security_package_versions = self.get_security_updates()
+
+        for index, package in enumerate(all_packages):
+            if package not in security_packages:
+                other_packages.append(package)
+                other_package_versions.append(all_package_versions[index])
+
+        self.composite_logger.log_debug("[TDNF] Discovered 'other' packages. [Count={0}]".format(len(other_packages)))
+        return other_packages, other_package_versions
```

Comment:
None of this code is useful (lot of busy work) in a scenario where security updates is always empty. You should just short circuit to returning the full list with an addendum to the function comment on why.

Reply:
Pending